### PR TITLE
AV-199140: Add default HTTP policy set for default-backend 404

### PR DESCRIPTION
**Pull request by ashishnayak96**
Adding a default backend "/" http policy set which will return 404 if the path is not met. This will also return 404 if there is no hostname match.
Case 1: Success

image

Case 2: Wrong path

image

Case 3: Wrong path but http route is "/" . This goes till the backend

image

Case 4: Wrong host

image

Case 5: Wildcard hostname

image

Pending Case:
Empty hostname

### CODE REVIEW

**Comment by ashishnayak96** [comment]
One of the test case is dependent on https://github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pull/1505

File: ako-gateway-api/nodes/avi_model_l7_translator.go [hunk 1]
```diff
@@ -31,6 +31,45 @@ import (
 	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pkg/utils"
 )
 
+func (o *AviObjectGraph) AddDefaultHTTPPS(key string) {
+	parentVS := o.GetAviEvhVS()[0]
+	defaultPSName := "default-backend"
```
Comment (aaha97) [inline ako-gateway-api/nodes/avi_model_l7_translator.go:36]: can we add this to the constants?

File: ako-gateway-api/nodes/avi_model_l7_translator.go [hunk 2]
```diff
@@ -31,6 +31,45 @@ import (
 	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pkg/utils"
 )
 
+func (o *AviObjectGraph) AddDefaultHTTPPS(key string) {
```
Comment (aaha97) [inline ako-gateway-api/nodes/avi_model_l7_translator.go:34]: can we rename the function to AddDefaultHTTPPolicySet for better readability

**Comment by ashishnayak96** [comment]
As discussed, will be returning 404 in case the hostname is not specified in http route and there is no path match

**Comment by ashishnayak96** [comment]
build ako

File: ako-gateway-api/nodes/avi_model_l7_translator.go (hunk 1 above)
Comment (ashishnayak96) [reply ako-gateway-api/nodes/avi_model_l7_translator.go]: Done

File: ako-gateway-api/nodes/avi_model_l7_translator.go (hunk 2 above)
Comment (ashishnayak96) [reply ako-gateway-api/nodes/avi_model_l7_translator.go]: Done

**Comment by ashishnayak96** [comment]
build ako

**Comment by ashishnayak96** [comment]
build ako

**Comment by ashishnayak96** [comment]
image

**Comment by ashishnayak96** [comment]
build ako

File: tests/gatewayapitests/graphlayer/httproute_test.go [hunk 3]
```diff
@@ -1173,7 +1173,7 @@ func TestHTTPRouteGatewayWithEmptyHostnameInGatewayHTTPRoute(t *testing.T) {
 	rules := []gatewayv1.HTTPRouteRule{rule}
 	hostnames := []gatewayv1.Hostname{}
 	akogatewayapitests.SetupHTTPRoute(t, httpRouteName, DEFAULT_NAMESPACE, parentRefs, hostnames, rules)
-
+	time.Sleep(2 * time.Second)
```
Comment (aaha97) [inline tests/gatewayapitests/graphlayer/httproute_test.go:1176]: why do we have this sleep?

Comment (ashishnayak96) [reply tests/gatewayapitests/graphlayer/httproute_test.go]: Done

**Comment by ashishnayak96** [comment]
image

**Comment by ashishnayak96** [comment]
build ako

**Review by aaha97 — APPROVED** [review]
LGTM

**Review by akshayhavile — APPROVED** [review]

**Comment by ashishnayak96** [comment]
build ako

**Review by pkoshtavmware — APPROVED** [review]
LGTM

**Pull request merged**